test/plots/areaPlotTests.ts

```diff
@@ -176,28 +195,17 @@ describe("Plots", () => {
 
         svg.remove();
       });
-
-      it("skips invalid Datasets", () => {
-        let twoPointDataset = new Plottable.Dataset([{ foo: 0, bar: 1 }, { foo: 1, bar: 2 }]);
-        areaPlot.addDataset(twoPointDataset);
-        let dummyDataset = new Plottable.Dataset([]);
-        let allAreas = areaPlot.selections([twoPointDataset, dummyDataset]);
-        assert.strictEqual(allAreas.size(), 2, "areas/lines retrieved");
-        let selectionData = allAreas.data()[0];
-        assert.deepEqual(selectionData, twoPointDataset.data(), "new dataset data in selection data");
-
-        svg.remove();
-      });
     });
 
-    it("retains original classes when class is projected", () => {
-      let newClassProjector = () => "pink";
-      areaPlot.attr("class", newClassProjector);
-      areaPlot.renderTo(svg);
-      let areaPath = renderArea.select(".area");
-      assert.isTrue(areaPath.classed("pink"));
-      assert.isTrue(areaPath.classed("area"));
-      svg.remove();
-    });
+    function getPointsInArea(plot: Plottable.Plots.Area<any>, points: Plottable.Point[]) {
+      let y0Binding = plot.y0();
+      let ds = plot.datasets()[0];
+      let bottomPoints = points.map((d, i) => {
+        let x = d.x;
+        let y0 = y0Binding.accessor(d, i, ds);
+        return { x: x, y: y0 };
+      }).reverse();
+      return points.concat(bottomPoints);
+    };
```

Comment:
well this function gets all points contained in the area, upper bounds and lower bounds
splitting into 2 functions or concatenating in the calling functions seems redundant and will make code harder to read ( in the invalid data tests for example)